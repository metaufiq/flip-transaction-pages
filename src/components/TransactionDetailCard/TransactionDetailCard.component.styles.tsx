import { StyleSheet } from "react-native";
import { colors } from "../../configs/themes";

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white'
  },
  idContainer: {
    padding: 20,
    borderBottomWidth: 1,
    borderBottomColor: '#f9f9f9'
  },
  detailHeaderContainer: {
    padding: 20,
    borderBottomWidth: 2,
    borderBottomColor: '#ebebeb',
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  bankContainer: {
    paddingTop: 20,
    paddingBottom: 10,
    paddingHorizontal: 20
  },
  rowContainer: {
    flexDirection:'row', 
    justifyContent:'space-between', 
    paddingVertical: 10,
    paddingHorizontal: 20,
  },
  id:{
    fontSize: 16,
    fontWeight: '600'
  },
  closeButton: {
    color: colors.primary,
    fontSize: 15,
    fontWeight: '600'
  }
})

export default styles;