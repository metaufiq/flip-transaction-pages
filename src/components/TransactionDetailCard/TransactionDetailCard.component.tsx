import React from "react";
import { Text, View } from "react-native";

import InformationDetail from "../InformationDetail";
import styles from "./TransactionDetailCard.component.styles";
import { Props } from "./TransactionDetailCard.component.types";

const TransactionDetailCard = ({transaction}: Props)=>{
  return(
    <View style={styles.container}>
      <Text>ID TRANSAKSI: #{transaction.id}</Text>
      <Text>DETAIL TRANSAKSI</Text>
      <Text>{transaction.sender_bank} â†’ {transaction.beneficiary_bank}</Text>
      <View>
        <InformationDetail header={transaction.beneficiary_name} content={transaction.account_number}/>
        <InformationDetail header={'NOMINAL'} content={transaction.amount}/>
        <InformationDetail header={'BERITA TRANSFER'} content={transaction.remark}/>
        <InformationDetail header={'KODE UNIK'} content={transaction.unique_code}/>
        <InformationDetail header={'WAKTU DIBUAT'} content={transaction.completed_at}/>
      </View>
      <View>
        <Text>{transaction.status}</Text>
      </View>
    </View>
  )
}

export default TransactionDetailCard;