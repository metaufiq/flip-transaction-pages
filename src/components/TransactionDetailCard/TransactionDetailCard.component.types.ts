import { Transaction } from "../../../index.types";

export interface Props{
  transaction: Transaction
}