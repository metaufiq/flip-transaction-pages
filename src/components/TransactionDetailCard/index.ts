import TransactionDetailCard from "./TransactionDetailCard.component";

export default TransactionDetailCard