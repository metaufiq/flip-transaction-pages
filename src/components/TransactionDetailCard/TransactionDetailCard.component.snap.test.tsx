import { assertSnapshots } from "../../../test";
import TransactionDetailCard from ".";
import { Props } from "./TransactionDetailCard.component.types";
import { Transaction } from "../../../index.types";

describe('TransactionDetailCardSnapshot', () => {
  const mockSuccessTransaction: Transaction = {
    account_number: 'account number',
    amount: 10000,
    beneficiary_bank: 'BENEF Bank',
    beneficiary_name: 'BENEF Name',
    completed_at: '07-07-2020',
    created_at: '07-07-2020',
    fee: 1000,
    id: '1231',
    remark: 'remark',
    sender_bank: 'SENDER Bank',
    status: 'SUCCESS',
    unique_code: 124
  };

  const mockPendingTransaction: Transaction = {
    account_number: 'account number',
    amount: 10000,
    beneficiary_bank: 'BENEF Bank',
    beneficiary_name: 'BENEF Name',
    completed_at: '07-07-2020',
    created_at: '07-07-2020',
    fee: 1000,
    id: '1231',
    remark: 'remark',
    sender_bank: 'SENDER Bank',
    status: 'PENDING',
    unique_code: 124
  };

  const mockProps:Props = {
    transaction: mockSuccessTransaction,
    onClose: ()=>{}
  };

  const configs = [
    {
      props: mockProps,
      desc: 'should render success transaction card'
    },
    {
      props: {...mockProps, transaction: mockPendingTransaction},
      desc: 'should render pending transaction card'
    }
  ];

  assertSnapshots(TransactionDetailCard, configs);
});
