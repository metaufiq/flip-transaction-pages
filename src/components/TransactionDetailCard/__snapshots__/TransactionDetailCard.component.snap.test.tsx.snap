// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionDetailCardSnapshot should render pending transaction card 1`] = `
<View
  style={Object {}}
>
  <Text>
    ID TRANSAKSI: #
    1231
  </Text>
  <Text>
    DETAIL TRANSAKSI
  </Text>
  <Text>
    SENDER Bank
     → 
    BENEF Bank
  </Text>
  <View>
    <View>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        BENEF Name
      </Text>
      <Text>
        account number
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        NOMINAL
      </Text>
      <Text>
        10000
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        BERITA TRANSFER
      </Text>
      <Text>
        remark
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        KODE UNIK
      </Text>
      <Text>
        124
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        WAKTU DIBUAT
      </Text>
      <Text>
        07-07-2020
      </Text>
    </View>
  </View>
  <View>
    <Text>
      PENDING
    </Text>
  </View>
</View>
`;

exports[`TransactionDetailCardSnapshot should render success transaction card 1`] = `
<View
  style={Object {}}
>
  <Text>
    ID TRANSAKSI: #
    1231
  </Text>
  <Text>
    DETAIL TRANSAKSI
  </Text>
  <Text>
    SENDER Bank
     → 
    BENEF Bank
  </Text>
  <View>
    <View>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        BENEF Name
      </Text>
      <Text>
        account number
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        NOMINAL
      </Text>
      <Text>
        10000
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        BERITA TRANSFER
      </Text>
      <Text>
        remark
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        KODE UNIK
      </Text>
      <Text>
        124
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        WAKTU DIBUAT
      </Text>
      <Text>
        07-07-2020
      </Text>
    </View>
  </View>
  <View>
    <Text>
      SUCCESS
    </Text>
  </View>
</View>
`;
